src/Product/ProductRepository.php

```diff
@@ -151,6 +151,22 @@ public function find_sync_ready_products( array $args = [], int $limit = - 1, in
 		return $this->product_filter->filter_sync_ready_products( $results, false );
 	}
 
+	/**
+	 * Find and return an array of WooCommerce product ID's ready to be deleted from the Google Merchant Center.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param int[] $ids    Array of WooCommerce product IDs
+	 * @param int   $limit  Maximum number of results to retrieve or -1 for unlimited.
+	 * @param int   $offset Amount to offset product results.
+	 *
+	 * @return array
+	 */
+	public function find_delete_product_ids( array $ids, int $limit = - 1, int $offset = 0 ): array {
```

Comment:
I looked at `ProductRepository` class, it has methods which do not only find, but also filter and there is a filter dependency added for three methods which are basically `find()` or `find_by_ids()`. Not sure I would do like this. I would think of `DeleteProducts` class as of 'controller' and call `ProductRepository` to find and then filter to filter out products by some criteria, not giving any filter functionality into the ProductRepository itself.

Reply:
It sounds like a great idea to separate the functionality, although I'm not sure how practical that would be. I'm not 100% clear on why it was implemented this way, but as far as I understand we need to find a balance between filtering, so it's best to do this at several levels.
The ProductRepository is responsible for filtering based on a DB query, we couldn't move all the filtering functionality into that class, because we would end up with some very expensive DB queries with multiple joins. Neither can we move all the filtering functionality to the ProductFilter class because then we would be retrieving the full product set, and looping through the products one by one, which for a site with 10k products would get excessive.
My intention with this PR was to stick as closely as possible to how it was done for syncing products, if you take a look at the Integration folder you will see several classes for compatible plugins, so I wanted to make it easier to use the a similar structure in case we needed to handle it differently for any of those classes.